h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -41,6 +41,8 @@ public static Vec toCategoricalVec(Vec src) {
         return numericToCategorical(src);
       case Vec.T_STR: // PUBDEV-2204
         return stringToCategorical(src);
+      case Vec.T_BAD:
+        return stringToCategorical(numericToStringVec(src));
```

Comment:
Why do we need to first convert to string and then to categorical? BAD Vecs are all NA, shouldnt numericToCategorical work just as well?

Reply:
good point, fixed